### DISCUSSION TITLE
Rollback latest successful dag code if dagrun has failed.

### DISCUSSION THREAD

**Discussion by starone99** [apache/airflow#34849] [Ideas]
If we click the dag, We can see the codebase in Web UI in airflow.
It seems that database is recording the code.

Then, how about saving the successful code and rolling back to latest successful code if dag run has failed?
I think this is useful on the production environment that dag run must not fail due to wrong code.
It should be optional since it can affect to other users or dag by dag.

**Comment by Taragolis** [comment]
The code in the UI shows only module code which are used during DAG Processor loop and this code doesn't used for execution.
Worker use actual code from the DAG directory as well as all installed dependencies.
